server/tools/readimage/tool.go

```
@@ -135,24 +146,41 @@ func (t *Tool) Call(ctx context.Context, c tools.Call) (tools.Result, error) {
 		return tools.ErrorResult(c, err.Error()), nil
 	}
 
-	info, err := os.Stat(resolvedPath)
+	file, info, err := openResolvedRegularFile(resolvedPath)
 	if err != nil {
-		return tools.ErrorResult(c, fmt.Sprintf("unable to locate file at %q: %v", resolvedPath, err)), nil
+		return tools.ErrorResult(c, err.Error()), nil
 	}
-	if !info.Mode().IsRegular() {
-		return tools.ErrorResult(c, fmt.Sprintf("path %q is not a regular file", resolvedPath)), nil
+	defer file.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `file.Close` is not checked

(errcheck)